server/BudgetTracker.Data/Repositories/Interfaces/IBudgetRepository.cs

```diff
@@ -49,5 +49,13 @@ public interface IBudgetRepository
         /// </p>
         /// </summary>
         Task<List<Budget>> GetRootBudgets(Guid userId);
+
+        /// <summary>
+        /// <p>
+        /// Fetches the entire tree of sub budgets for the given root budget and
+        /// attaches them to that root budget's SubBudgets property.
+        /// </p>
+        /// </summary>
+        Task LoadBudgetTree(Budget rootBudget);
```

Comment:
Are we missing the implementation for this method?

Reply:
This is a draft PR. So at the moment, yes the implementation doesn't exist. I'm still working on it.